api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -371,6 +371,11 @@ message HttpConnectionManager {
   ServerHeaderTransformation server_header_transformation = 34
       [(validate.rules).enum = {defined_only: true}];
 
+  // By default, Envoy removes hop by hop headers. For envoy-mobile
+  // they should be passed up to the caller. This should not be used by server
+  // side Envoy as it expects hop by hop headers to be cleared.
+  bool preserve_hop_by_hop_response_headers = 48;
```

Comment:
Thinking out loud here, are there going to be more mobile-only fields? If so, should this be part of a sub message that makes it clear that this is only for envoy-mobile instances?
This way we can also reject a config that has this sub-message in non-envoy-mobile instances.

Reply:
Also for consideration, can we make HCM aware of, in some general sense, the fact that it's linked into a client and not another hop? I think to reject and validate this field on Envoy-proper, we would need this kind of awareness anyway.